diff --git a/node_modules/react-native-safe-area-context/android/build.gradle b/node_modules/react-native-safe-area-context/android/build.gradle
index e556043..7ef7403 100644
--- a/node_modules/react-native-safe-area-context/android/build.gradle
+++ b/node_modules/react-native-safe-area-context/android/build.gradle
@@ -70,6 +70,9 @@ android {
     lintOptions{
         abortOnError false
     }
+    buildFeatures {
+        buildConfig true
+    }
 
     packagingOptions {
         // For some reason gradle only complains about the duplicated version of libreact_render libraries
